### PR TITLE
fix(spanner): upgrade spanner client to latest to support NUMERIC types

### PR DESCRIPTION
**Pull request by kylechadha**
Update spanner library to latest, `cloud.google.com/go/spanner@v1.18.0`

This prevents `error: :1.57: got "NUMERIC", want scalar type`

Fixes: https://github.com/golang-migrate/migrate/issues/564

### CODE REVIEW

File: database/spanner/spanner_test.go [hunk 1]
```diff
@@ -151,6 +151,14 @@ func TestCleanStatements(t *testing.T) {
 			SET OPTIONS (allow_commit_timestamp=true);`,
 			expected: []string{"ALTER TABLE users ALTER COLUMN created SET OPTIONS (allow_commit_timestamp = true)"},
 		},
+		{
+			name: "column with NUMERIC type",
```
Comment (dhui) [inline database/spanner/spanner_test.go:155]: This only tests that `cleanStatements()` passes. It doesn't test running the statement against the emulator. Was this your intention?

**Review by dhui — DISMISSED** [review]
Thanks for reporting the issue and fixing it!

File: database/spanner/spanner_test.go (hunk 1 above)
Comment (kylechadha) [reply database/spanner/spanner_test.go:155]: Whoops, just updated where the last spanner upgrade PR did https://github.com/golang-migrate/migrate/pull/427 -- will add a migration to test this case as well. Let me know if there's anywhere else that should be updated!

**Comment by kylechadha** [comment]
@dhui Looks like the Spanner tests pass -- Not sure why the Cassandra migrations would have failed. Possible they're flappy?

File: database/spanner/spanner_test.go (hunk 1 above)
Comment (dhui) [reply database/spanner/spanner_test.go:155]: If DDL parsing is all you want to test, then this is fine. But if you want to test table creation, then this doesn't test that. I'm fine either way.

**Review by dhui — APPROVED** [review]

File: database/spanner/spanner_test.go (hunk 1 above)
Comment (kylechadha) [reply database/spanner/spanner_test.go:155]: Hi @dhui not sure I understand, when you say 'these examples' are you saying the tests are good as is, or that I should remove the migrations and add a new test function? If you have an example of what you're referring to that would be great

**Pull request merged**